test/jsupdate-behaviour-mocha.js

```diff
@@ -112,10 +213,115 @@ describe("framework-state", function() {
+                });
+            }).catch(fail);
+        }).should.eventually.have.property('data', "Hello again");
+    });
+    it("should fire updater if upstream updater changed output state lm", function() {
+        return new Promise(function(done, fail){
+            var count = 0;
+            test.createNetwork({
+                once: jswrapper(function(input) {
+                    switch(++count) {
+                        case 1: return input;
+                        default:
+                        var outputState = this.outputState();
+                        outputState.lm = "changed";
+                        this.outputState(outputState);
+                    }
+                }),
+                sut: jswrapper(function(input) {
+                    return "Hello " + input;
+                })
+            }).then(function(network){
+                network.graph.addEdge('once', 'output', 'sut', 'input');
+                return new Promise(function(adv) {
+                    test.onOutPortData(network.processes.sut.component, 'output', adv);
+                    test.sendData(network.processes.once.component, 'input', "once");
+                }).then(function() {
+                    test.onOutPortData(network.processes.sut.component, 'output', done);
+                    test.sendData(network.processes.once.component, 'input', "again");
+                });
+            }).catch(fail);
+        }).should.eventually.have.property('data', "Hello once");
+    });
+    it("should fire updater if upstream updater did nothing, after error, after valid output", function() {
+        return new Promise(function(done, fail){
+            var count = 0;
+            test.createNetwork({
+                upstream: jswrapper(function(input) {
+                    switch(++count) {
+                        case 1: return input;
+                        case 2: throw input;
+                        case 3: return;
+                    }
+                }),
+                sut: jswrapper(function(input) {
+                    return "Hello " + input;
+                })
+            }).then(function(network){
+                network.graph.addEdge('upstream', 'output', 'sut', 'input');
+                return new Promise(function(adv) {
+                    test.onOutPortData(network.processes.sut.component, 'output', adv);
+                    test.sendData(network.processes.upstream.component, 'input', "upstream");
+                }).then(function() {
+                    return new Promise(function(adv) {
+                        test.onOutPortData(network.processes.upstream.component, 'output', adv);
+                        test.sendData(network.processes.upstream.component, 'input', "again");
+                    });
+                }).then(function() {
+                    test.onOutPortData(network.processes.sut.component, 'output', done);
```

Comment:
I am probably just not following the code properly, but I do not understand how 'done' would be called at this point if the sut updater does not fire, and at this point the sut updater should not fire.   If the sut updater does not fire, then its output listeners should not be notified -- there should not be an output event for sut's output -- so what would cause 'done' on line 294 to be invoked?